### PR TITLE
fix(kit): emit null from mode provider instead cancel stream

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by waterplea — CHANGES_REQUESTED** [review]
I think it makes more sense to change `EMPTY` to `of(null)` in `MODE_PROVIDER`

File: projects/kit/components/input-tag/input-tag.component.ts [hunk 1]
```diff
@@ -295,12 +295,8 @@ export class TuiInputTagComponent
         return this.hasCleaner || this.hasTooltip || this.iconAlignRight;
     }
 
-    /**
-     * @deprecated: use `status$ | async` instead
-     * TODO: remove in v3.0
-     */
     get status(): TuiStatusT {
-        return this.modeDirective && this.modeDirective.mode ? 'default' : this.tagStatus;
+        return this.modeDirective?.mode ? 'default' : this.tagStatus;
```
Comment (nsbarsukov) [inline projects/kit/components/input-tag/input-tag.component.ts:299]: we have 2 variables about status in different parts of file.

Can we place this logic inside stream `status$` (and drop this getter) ?
Something like this
```ts
status$: Observable<TuiStatusT> = this.mode$.pipe(
        map(mode => (mode ? 'default' : this.tagStatus)),
        map(status => status || this.modeDirective?.mode ? 'default' : this.tagStatus)
    );
```

Comment (splincode) [reply projects/kit/components/input-tag/input-tag.component.ts:299]: it's probably async pipe is never emitted and return null

Comment (splincode) [reply projects/kit/components/input-tag/input-tag.component.ts:299]: I changed to 

```
status$: Observable<TuiStatusT> = this.mode$.pipe(map(() => this.status));
```

**Review by waterplea — APPROVED** [review]

**Comment by splincode** [comment]
@waterplea @nsbarsukov sorry, updated

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**